client/state/reader/follows/reducer.js

```diff
@@ -11,10 +11,83 @@ import {
 	READER_FOLLOW,
 	READER_UNFOLLOW,
 	READER_FOLLOWS_RECEIVE,
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL,
 } from 'state/action-types';
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
+function updatePostSubscription( state, { payload, type } ) {
+	const follow = find( state, { blog_ID: +payload.blogId } );
+	if ( ! follow ) {
+		return state;
+	}
+
+	const currentFollowState = get( follow, [ 'delivery_methods', 'email' ], {} );
+
+	const newProps = {
+		send_posts: ! ( type === READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL )
+	};
+
+	if ( payload.deliveryFrequency ) {
+		newProps.post_delivery_frequency = payload.deliveryFrequency;
+	}
+
+	const newFollowState = {
+		...currentFollowState,
+		...newProps,
+	};
+
+	if ( isEqual( currentFollowState, newFollowState ) ) {
```

Comment:
what is the case we are guarding from? when could there be an action that ends up creating a new equal follow state?